fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/capture/FingerprintCaptureWrapperV2.kt

```diff
@@ -12,159 +12,94 @@ import com.simprints.fingerprint.infra.scanner.v2.domain.main.message.un20.model
+    }
 
     override suspend fun acquireFingerprintImage(): AcquireFingerprintImageResponse =
-        withContext(ioDispatcher) {
-            scannerV2.acquireImage(IMAGE_FORMAT).map { imageBytes ->
-                AcquireFingerprintImageResponse(imageBytes.image)
-            }
-                .switchIfEmpty(Single.error(NoFingerDetectedException("Failed to acquire image")))
-                .wrapErrorsFromScanner()
-                .await()
+        executeSafely {
+            scannerV2.acquireImage(IMAGE_FORMAT)?.image?.let { imageBytes ->
+                AcquireFingerprintImageResponse(imageBytes)
+            } ?: throw NoFingerDetectedException("Failed to acquire image")
         }
 
 
     override suspend fun acquireUnprocessedImage(
         captureDpi: Dpi?,
-    ): AcquireUnprocessedImageResponse =
-
-        withContext(ioDispatcher) {
-            require(captureDpi != null && (captureDpi.value in MIN_CAPTURE_DPI..MAX_CAPTURE_DPI)) {
-                "Capture DPI must be between $MIN_CAPTURE_DPI and $MAX_CAPTURE_DPI"
-            }
-            // Capture fingerprint and ensure it's OK
-            scannerV2.captureFingerprint()
-                .ensureCaptureResultOkOrError()
-                .andThen(Completable.fromAction {
-                    tracker.completeScan()
-                }).andThen(acquireUnprocessedImage())
-                .switchIfEmpty(Single.error(NoFingerDetectedException("Failed to acquire unprocessed image data")))
-                .wrapErrorsFromScanner().await()
-
+    ): AcquireUnprocessedImageResponse = executeSafely {
+        require(captureDpi != null && (captureDpi.value in MIN_CAPTURE_DPI..MAX_CAPTURE_DPI)) {
+            "Capture DPI must be between $MIN_CAPTURE_DPI and $MAX_CAPTURE_DPI"
         }
-
-
-    private fun acquireUnprocessedImage() =
-        scannerV2.acquireUnprocessedImage(IMAGE_FORMAT)
-            .map { imageData ->
-                AcquireUnprocessedImageResponse(
-                    RawUnprocessedImage(
-                        imageData.image
-                    )
+        // Capture fingerprint and ensure it's OK
+        scannerV2.captureFingerprint().ensureCaptureResultOkOrError()
+        tracker.completeScan()
+        // Transfer the unprocessed image from the scanner
+        scannerV2.acquireUnprocessedImage(IMAGE_FORMAT)?.image?.let { imageData ->
+            AcquireUnprocessedImageResponse(
+                RawUnprocessedImage(
+                    imageData
                 )
-            }
+            )
+        } ?: throw NoFingerDetectedException("Failed to acquire unprocessed image data")
+    }
 
     override suspend fun acquireFingerprintTemplate(
-        captureDpi: Dpi?,
-        timeOutMs: Int,
-        qualityThreshold: Int,
-        allowLowQualityExtraction: Boolean
-    ): AcquireFingerprintTemplateResponse = withContext(ioDispatcher) {
+        captureDpi: Dpi?, timeOutMs: Int, qualityThreshold: Int, allowLowQualityExtraction: Boolean
```

Comment:
Why did you put parameters on the same line? (this actually applies to many other places, but I don't want to pollute the PR with too many styling comments)